daml-assistant/exe/DAML/Assistant.hs

```diff
@@ -25,7 +26,9 @@ import Control.Monad.Extra
 -- | Run the assistant and exit.
 main :: IO ()
 main = displayErrors $ do
-    env@Env{..} <- getDamlEnv
+    rawArgs <- getArgs
+    let isInstall = listToMaybe rawArgs == Just "install"
```

Comment:
```suggestion
    let isInstall = ["install"] `isPrefixOf` rawArgs
```

Pretty much code golf at this point, but i think this version reads semantically nicer.

Reply:
I'm not going to use this suggestion because I think the test above is easier to generalize to more commands that are already planned (e.g. `daml update` and `daml uninstall`, not yet implemented).

Reply:
Given it's a bit of a hack, and you plan to generalise it using it more, any reason not to just rerun the command line parser here to determine what type of command it is?

Reply:
You mean, run it for the first time here, and run it again later with more info.

Part of the problem is that determining the full set of available commands requires knowing the SDK version and the SDK config. The problem comes when we use optparse-applicative to generate `--help` tables, for example.

I think this can all be refactored and made proper with optparse-applicative, but it's not trivial and still requires two passes over the command-line. Not sure it's worth it over the simple solution...

Reply:
I struggle to believe there's not a command line you'll misparse if you write the parsing logic twice. But perhaps anyone who writes "odd" command lines deserves the pain. I leave it to you.